lib/fetchHttpClient.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import "isomorphic-fetch";
+import * as tough from "tough-cookie";
+import AbortController from "abort-controller";
+import FormData from "form-data";
+
+import { HttpClient } from "./httpClient";
+import { WebResource } from "./webResource";
+import { HttpOperationResponse } from "./httpOperationResponse";
+import { HttpHeaders } from "./httpHeaders";
+import { RestError } from "./restError";
+import { Readable, Transform } from "stream";
+
+interface FetchError extends Error {
+    code: string | undefined;
+    errno: string | undefined;
+    type: string | undefined;
+}
+
+export class FetchHttpClient implements HttpClient {
+    private readonly cookieJar = new tough.CookieJar();
```

Comment:
is tough-cookie isomorphic? I know, there were problems with it previously.

Reply:
This is for Node-only right now